### ISSUE TITLE
which one is right about RedirAddr or ServerName?

### ISSUE THREAD

**Issue by qwerttvv** [cbeuw/Cloak#89]
**or**
which one is **better** about RedirAddr or ServerName?

princeton.edu 140.180.223.22

www.princeton.edu 140.180.223.42

![image](https://user-images.githubusercontent.com/56712217/72303769-4bf5e300-36a9-11ea-8aa9-6ae5e675fa82.png)

![image](https://user-images.githubusercontent.com/56712217/72303692-09340b00-36a9-11ea-80ac-61737c5cdfe6.png)

**Comment by malikshi** [comment]
redirAddr for serverside you can use IP/hostname. 
ServerName for clientside using hostname only

**Comment by qwerttvv** [comment]
> redirAddr for serverside you can use IP/hostname.
> ServerName for clientside using hostname only
> see example: https://github.com/cbeuw/Cloak/tree/master/example_config

www.bing.com 204.79.197.200????

In china, i dont think the example is a good idea……

![image](https://user-images.githubusercontent.com/56712217/72305023-6c27a100-36ad-11ea-9c39-fd08d39e6103.png)

![image](https://user-images.githubusercontent.com/56712217/72305058-85305200-36ad-11ea-9569-5f8c4fb3721e.png)

**Comment by malikshi** [comment]
You can use any hostname as long as work in you country, I just mention example only.
Usually I'm using IP of my own isp that have collaboration with hostname like line/fb/etc for serversidw and use ServerName of fb/line in client side

**Comment by klaaktu** [comment]
`bing.com` goes to `13.107.21.200`/`204.79.197.200`, which are non-cn IPs. And then they redirect to `www.bing.com`, or `cn.bing.com` if accessed by Chinese IP. `13.107.21.200`/`204.79.197.200` & `202.89.233.100`/`202.89.233.101` seem to be MS CDNs (they all serve both versions). Which IP u get depend on DNS response.
For example my ISP DNS says `bing.com` goes to `204.79.197.200`/`13.107.21.200`, then `cn.bing.com` to `202.89.233.100`/`202.89.233.101`, I guess MS got the IPs/domain whitelisted by GFW to avoid problems.

**Comment by malikshi** [comment]
Yes. You can choose one of them that got white listed gfw. I think they didn't change that often so it's would be fine. I prefer using IPs than using hostname(I think latest update support redir on IPs and hostname) on side server

**Comment by qwerttvv** [comment]
now, i use 
ServerName: www.princeton.edu
RedirAddr: www.princeton.edu

i think princeton cant be block by gfw....its a university......

**Comment by klaaktu** [comment]
Fail to resolve RedirAddr causes ck-server to fail
```
ck-server-linux-amd64[666]: time="2021-04-08T13:02:11Z" level=fatal msg="unable to initialise server state: unable to parse RedirAddr: unable to resolve RedirAddr: lookup www.bing.com on 127.0.0.1:53: read udp 127.0.0.1:56266->127.0.0.1:53: i/o timeout. "

```
